packages/components/tree-v2/src/tree-node.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <div
+    ref="node$"
+    class="el-tree-node"
+    :class="{
+      'is-expanded': expanded,
+      'is-current': current,
+      'is-focusable': !disabled,
+      'is-checked': !disabled && checked,
+    }"
+    role="treeitem"
+    tabindex="-1"
+    :aria-expanded="expanded"
+    :aria-disabled="disabled"
+    :aria-checked="checked"
+    :data-key="node.key"
+    @click.stop="handleClick"
+    @contextmenu="handleContextMenu"
+  >
+    <div
+      class="el-tree-node__content"
+      :style="{ paddingLeft: (node.level - 1) * indent + 'px' }"
+    >
+      <span
+        :class="[
+          {
+            'is-leaf': node.isLeaf,
+            'is-hidden': hiddenExpandIcon,
+            expanded: !node.isLeaf && expanded,
+          },
+          'el-tree-node__expand-icon',
+          icon,
+        ]"
+        @click.stop="handleExpandIconClick"
+      >
+      </span>
+      <el-checkbox
+        v-if="showCheckbox"
+        :model-value="checked"
+        :indeterminate="indeterminate"
+        :disabled="disabled"
+        @change="handleCheckChange"
+        @click.stop
+      />
+      <el-node-content :node="node" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, inject } from 'vue'
+import ElCheckbox from '@element-plus/components/checkbox'
+import ElNodeContent from './tree-node-content.vue'
+import { treeNodeProps } from './defaults'
+import { RootTreeType } from './tree.type'
+
+const DEFAULT_ICON = 'el-icon-caret-right'
+
+export default defineComponent({
+  name: 'ElTreeNode',
+  components: {
+    ElCheckbox,
+    ElNodeContent,
+  },
+  props: treeNodeProps,
+  emits: ['click', 'toggle', 'check'],
+  setup(props, { emit }) {
+    const tree = inject<RootTreeType>('RootTree')
+
+    const indent = computed(() => {
+      return tree.props.indent
+    })
+
+    const icon = computed(() => {
+      return tree.props.iconClass ? tree.props.iconClass : DEFAULT_ICON
+    })
```

Comment:
We should now use `ElIcon` with `svg` icon instead of using the old way.
We do not want to introduce more legacy code in new components. Please follow the instruction to add new icon.